src/wallet/view_seed_flow.c

```diff
@@ -0,0 +1,150 @@
+ * limitation fees for hosting or consulting/ support services related
+ * to the Software), a product or service whose value derives, entirely
+ * or substantially, from the functionality of the Software. Any license
+ * notice or attribution required by the License must also include
+ * this Commons Clause License Condition notice.
+ *
+ * Software: All X1Wallet associated files.
+ * License: MIT
+ * Licensor: HODL TECH PTE LTD
+ *
+ ******************************************************************************
+ */
+
+/*****************************************************************************
+ * INCLUDES
+ *****************************************************************************/
+#include "constant_texts.h"
+#include "core_error.h"
+#include "reconstruct_wallet_flow.h"
+#include "ui_delay.h"
+#include "ui_list.h"
+#include "ui_multi_instruction.h"
+#include "ui_state_machine.h"
+/*****************************************************************************
+ * EXTERN VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE MACROS AND DEFINES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE TYPEDEFS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTION PROTOTYPES
+ *****************************************************************************/
+/**
+ * @brief The function handles transitions of view seed flow for the passed
+ * wallet id.
+ *
+ * @param wallet_id A pointer to a uint8_t array representing the wallet ID.
+ *
+ * @return true if view seed is successful, else false.
+ */
+static bool view_seed_handler(const uint8_t *wallet_id);
+
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * GLOBAL VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTIONS
+ *****************************************************************************/
+static bool view_seed_handler(const uint8_t *wallet_id) {
+  CONFIDENTIAL char mnemonics[MAX_NUMBER_OF_MNEMONIC_WORDS]
+                             [MAX_MNEMONIC_WORD_LENGTH] = {0};
+  CONFIDENTIAL uint8_t no_of_mnemonics = 0;
+  bool result = false;
+
+  do {
+    multi_instruction_init(
+        ui_text_view_seed_messages, 3, DELAY_LONG_STRING, true);
+    if (0 != get_state_on_confirm_scr(0, 1, 2)) {
+      break;
+    }
```

Comment:
Reconsider replacing this with confirmation screen instead..

Reply:
updated